src/v0/destinations/google_adwords_remarketing_lists/transformV2.js

```diff
@@ -0,0 +1,287 @@
+    customerId: filteredCustomerId,
+    consent: consentBlock,
+  };
+  response.headers = {
+    Authorization: `Bearer ${accessToken}`,
+    'Content-Type': JSON_MIME_TYPE,
+    'developer-token': developerToken,
+  };
+  if (Config.subAccount)
+    if (Config.loginCustomerId) {
+      const filteredLoginCustomerId = removeHyphens(Config.loginCustomerId);
+      response.headers['login-customer-id'] = filteredLoginCustomerId;
+    } else throw new ConfigurationError(`loginCustomerId is required as subAccount is true.`);
+  return response;
+};
+
+const populateIdentifiers = (attributeArray, { Config }) => {
+  const userIdentifier = [];
+  const { typeOfList } = Config;
+  const { isHashRequired, userSchema } = Config;
+  let attribute;
+  if (TYPEOFLIST[typeOfList]) {
+    attribute = TYPEOFLIST[typeOfList];
+  } else {
+    attribute = userSchema;
+  }
+  if (isDefinedAndNotNullAndNotEmpty(attributeArray)) {
+    // traversing through every element in the add array
+    attributeArray.forEach((element, index) => {
+      if (isHashRequired) {
+        hashEncrypt(element);
+      }
+      // checking if the attribute is an array or not for generic type list
+      if (!Array.isArray(attribute)) {
+        if (element[attribute]) {
+          userIdentifier.push({ [attribute]: element[attribute] });
+        } else {
+          logger.info(` ${attribute} is not present in index:`, index);
+        }
+      } else {
+        attribute.forEach((attributeElement, index2) => {
+          if (attributeElement === 'addressInfo') {
+            const addressInfo = constructPayload(element, addressInfoMapping);
+            // checking if addressInfo object is empty or not.
+            if (isDefinedAndNotNullAndNotEmpty(addressInfo)) userIdentifier.push({ addressInfo });
+          } else if (element[`${attributeElement}`]) {
+            userIdentifier.push({
+              [`${attributeMapping[attributeElement]}`]: element[`${attributeElement}`],
+            });
+          } else {
+            logger.info(` ${attribute[index2]} is not present in index:`, index);
+          }
+        });
+      }
+    });
+  }
+  return userIdentifier;
+};
+
+const processRecordEventArray = (
+  records,
+  message,
+  destination,
+  accessToken,
+  developerToken,
+  operationType,
+) => {
+  let outputPayloads = {};
+  // ** only send it if identifier > 0
+
+  const fieldsArray = [];
+  const metadata = [];
+  records.forEach((record) => {
+    fieldsArray.push(record.message.fields);
+    metadata.push(record.metadata);
+  });
+
+  const userIdentifiersList = populateIdentifiers(fieldsArray, destination);
+
+  const outputPayload = constructPayload(message, offlineDataJobsMapping);
+  outputPayload.operations = [];
+  // breaking the userIdentiFier array in chunks of 20
+  const userIdentifierChunks = returnArrayOfSubarrays(userIdentifiersList, 20);
+  // putting each chunk in different create/remove operations
+  switch (operationType) {
```

Comment:
We need to consider the update opertationType as well.

Reply:
for update, we are sending 'add' in the fucntion call, updates will be treated as adds